src/main/java/utils/HttpRequestPath.java

```diff
@@ -0,0 +1,57 @@
+package utils;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class HttpRequestPath {
+    private static final String PATH_PARSER_TOKEN_WITH_ESCAPE = "\\?";
+    private static final String PATH_PARSER_TOKEN = "?";
+    private static final String QUERY_STRING_PARSER_TOKEN = "&";
+    private static final String KEY_VALUE_PARSER_TOKEN = "=";
+
+    private final String path;
+    private final HttpRequestQueryString httpRequestQueryString;
+
+    public HttpRequestPath(String path, HttpRequestQueryString httpRequestQueryString) {
+        this.path = path;
+        this.httpRequestQueryString = httpRequestQueryString;
+    }
+
+    public static HttpRequestPath findHttpRequestPath(String pathString) {
+        Map<String, String> httpRequestQueryStrings
+            = new HashMap<>();
+
+        String[] split = pathString.split(PATH_PARSER_TOKEN_WITH_ESCAPE);
+
+        if (pathString.contains(PATH_PARSER_TOKEN)) {
+            String[] querySplit = split[1].split(QUERY_STRING_PARSER_TOKEN);
+
+            httpRequestQueryStrings = Arrays.stream(querySplit)
+                .map(parameter -> parameter.split(KEY_VALUE_PARSER_TOKEN))
+                .collect(Collectors.toMap(
+                    keyValue -> keyValue[0], keyValue -> keyValue[1]
+                ));
+        }
+
+        return new HttpRequestPath(split[0], new HttpRequestQueryString(httpRequestQueryStrings));
```

Comment:
아래와 같이 변경해볼 수 있을 것 같아요. 😉
```suggestion
        public static HttpRequestPath findHttpRequestPath(String pathString) {
        String[] pathComponents = pathString.split(PATH_PARSER_TOKEN_WITH_ESCAPE);

        if (pathString.contains(PATH_PARSER_TOKEN)) {
            String[] querySplit = pathComponents[1].split(QUERY_STRING_PARSER_TOKEN);

            Map<String, String> httpRequestQueryStrings = Arrays.stream(querySplit)
                .map(parameter -> parameter.split(KEY_VALUE_PARSER_TOKEN))
                .collect(Collectors.toMap(
                    keyValue -> keyValue[0], keyValue -> keyValue[1]
                ));

            return new HttpRequestPath(pathComponents[0], new HttpRequestQueryString(httpRequestQueryStrings));
        }

        return new HttpRequestPath(pathComponents[0], HttpRequestQueryString.empty());
    }
```